huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go

```diff
@@ -763,7 +775,9 @@ func resourceCBHInstanceRead(_ context.Context, d *schema.ResourceData, meta int
 		if publicIpId != nil && strings.TrimSpace(publicIpId.(string)) != "" {
 			publicIp, err = getPublicAddressById(d, cfg, strings.TrimSpace(publicIpId.(string)))
 			if err != nil {
-				return diag.FromErr(err)
+				if _, ok := err.(golangsdk.ErrDefault404); !ok {
+					return diag.FromErr(err)
```

Comment:
should we clear **publicIpId** when the error code is 404?

Reply:
done